src/sentry/static/sentry/app/views/performance/realUserMonitoring/index.tsx

```diff
@@ -0,0 +1,177 @@
+
+        browserHistory.replace({
+          pathname: `/organizations/${organization.slug}/performance/`,
+          query: {
+            ...location.query,
+          },
+        });
+        return null;
+      }
+
+      return (
+        <Redirect
+          router={router}
+          to={transactionSummaryRouteWithQuery({
+            orgSlug: organization.slug,
+            transaction: transactionName,
+            projectID: decodeScalar(location.query.project),
+            query: location.query,
+          })}
+        />
+      );
+    } else {
+      return <Alert type="warning">{t("You don't have access to this feature")}</Alert>;
+    }
+  };
+
+  render() {
+    const {organization, projects, location} = this.props;
+    const {eventView} = this.state;
+    const transactionName = getTransactionName(location);
+    if (!eventView || transactionName === undefined) {
+      // If there is no transaction name, redirect to the Performance landing page
+      browserHistory.replace({
+        pathname: `/organizations/${organization.slug}/performance/`,
+        query: {
+          ...location.query,
+        },
+      });
+      return null;
+    }
+
+    return (
+      <SentryDocumentTitle title={this.getDocumentTitle()} objSlug={organization.slug}>
+        <Feature
+          features={['measurements']}
+          organization={organization}
+          renderDisabled={this.renderNoAccess}
+        >
+          <GlobalSelectionHeader>
+            <StyledPageContent>
+              <LightWeightNoProjectMessage organization={organization}>
+                <RumContent
+                  location={location}
+                  eventView={eventView!}
+                  transactionName={transactionName!}
+                  organization={organization}
+                  projects={projects}
+                />
+              </LightWeightNoProjectMessage>
+            </StyledPageContent>
+          </GlobalSelectionHeader>
+        </Feature>
+      </SentryDocumentTitle>
+    );
+  }
+}
+
+const StyledPageContent = styled(PageContent)`
+  padding: 0;
+`;
+
+function generateRumEventView(
+  location: Location,
+  transactionName: string | undefined
+): EventView | undefined {
+  if (transactionName === undefined) {
+    return undefined;
+  }
+  const query = decodeScalar(location.query.query) || '';
+  const conditions = tokenizeSearch(query);
+  conditions
+    .setTag('event.type', ['transaction'])
+    .setTag('transaction.op', ['pageload'])
+    .setTag('transaction', [transactionName]);
+
+  Object.keys(conditions.tagValues).forEach(field => {
+    if (isAggregateField(field)) conditions.removeTag(field);
+  });
+
+  return EventView.fromNewQueryWithLocation(
+    {
+      id: undefined,
+      version: 2,
+      name: transactionName,
+      fields: ['id', 'user', 'transaction.duration', 'timestamp'],
+      query: stringifyQueryObject(conditions),
+      projects: [],
```

Comment:
since you are passing an empty array for projects, doesn't it just get overridden with projects from location? should you pass the projects from props to be more explicit?

Reply:
We want it to be overridden with projects from the location as it should just be filtering on that one project (the transaction summary page works the same way). The projects from the props is a list of all projects that's used by the `Create Alert` button in the header.

Reply:
ah 👍